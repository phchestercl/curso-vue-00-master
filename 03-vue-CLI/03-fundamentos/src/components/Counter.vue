<template>
  <div>
    <h2>{{ customTitle }}</h2>
    <p>{{ counter }} <sup>2</sup> {{ counter * counter }}</p>
    <p data-test-id="counter"> {{ counter}} </p>
    <div class="buttons-container">
        <button v-on:click="sumaUno">+1</button>
        <button @click="restaUno">-1</button>
        </div>
  </div>
</template>

<script>
export default {
  name: "Counter",
  /* props:['titulo-de-la-seccion','start'], */
  props:{
    tituloDeLaSeccion:String,
    start:{
      type:Number,
      default: 100,
      /* required: true, */
      validator(value){
        return value>=0
      }
    }
  },
  data() {
    return {
      counter: this.start,
    };
  },
  methods: {
    getSquareValue() {
      console.log("getSquareValue");
      return this.counter * this.counter;
    },
    sumaUno(){
        return this.counter+=1
    },
    restaUno(){
        this.counter-=1
    }
  },
  computed: {
    squareCounter() {
      console.log("computed squareCounter");
      return this.counter * this.counter;
    },
    customTitle(){
        return this.tituloDeLaSeccion||'Counter'
        /* if (!this.tituloDeLaSeccion) {
            return'Counter'
        } else {
            return this.tituloDeLaSeccion
        } */
    }
    
  },
};
</script>

<style>
button {
  background-color: #358f5c;
  border-radius: 4px;
  border: 1px solid whitesmoke;
  color: whitesmoke;
  cursor: pointer;
  margin: 5px;
  padding: 4px;
  transition: 0.3s ease-in-out;
  width:60px
}
button:hover{
    background-color: #224933;
    transition: 0.3s ease-in-out;
    color: whitesmoke;

}
</style>