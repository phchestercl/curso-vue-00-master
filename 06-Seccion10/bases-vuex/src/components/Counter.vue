<template>
<div>
    <h1>Counter - Vuex</h1>
    <!-- <h2>Direct access: {{ $store.state.count }}</h2> -->
    <h2>Direct access: {{ $store.state.counter.count }}</h2>

    <h2>Computed : {{countComputed}}</h2>
        <button @click="increment">+1</button>
        <button @click="incrementByFive">+5</button>
       <!--  <button @click="randInc">+Random</button> -->
       <button @click="incrementRandomInt"
        :disabled="isLoading"
        >+Random</button>
    <h2>MapState</h2>
    <h2>mapState : {{count}} </h2>
    <h2>lastMutation : {{lastMutation}} </h2>
    <h2>Direct getters : {{ squareCount }} </h2>

</div>
</template>

<script>

import { mapState,mapActions, mapGetters } from "vuex";
console.log({ mapState });
export default {
  computed: {
    countComputed() {
        //return this.$store.state.count; 
        return this.$store.state.counter.count; // Para llamadas del store counter
    },
    ...mapState('counter',["count", "lastMutation",'isLoading']),
    /* ...mapState({
           count: state=>state.count,
           lastMutation: state=>state.lastMutation
       }) */
    ...mapGetters('counter',['squareCount'])
  },
  methods: {
    increment() {
      //this.$store.commit("increment")  // => para llamadas del store global
      this.$store.commit("counter/increment") //==>> para llamadas del store del modulo counter
    },
    incrementByFive() {
      this.$store.commit("counter/incrementBy", 5)
      // si en lugar de utizar esta función quisieramos ejecutar una accion
      // definida en em mapActions, tendríamos que llamarla de la siguiente manera
      // this.incrementRandomInt()
    },
    // ...mapActions(['incrementRandomInt']) // si no hubiera un mudulo utilizariamos 
    //...mapActions('counter',['incrementRandomInt'])
    ...mapActions('counter',{
        incrementRandomInt:'incrementRandomInt'
    })
    /* randInc(){
        //este componente no tiene mucho sentido ya que lo unico que
        // es llamar una actions
        this.$store.dispatch('incrementRandomInt')
    } */
  },
};
</script>

<style>
</style>