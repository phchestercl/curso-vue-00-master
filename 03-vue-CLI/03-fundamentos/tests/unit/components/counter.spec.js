import { shallowMount, mount } from '@vue/test-utils'
import Counter from '@/components/Counter'

describe('Counter Component', ()=>{
    // Esta prueba nos permite saber si el cÃ³digo se encuentra tal como lo dejamos
    test('Debe ser identico al Snapshot',()=>{
        const wrapper = shallowMount( Counter)
        
        expect( wrapper.html() ).toMatchSnapshot()

    })
})