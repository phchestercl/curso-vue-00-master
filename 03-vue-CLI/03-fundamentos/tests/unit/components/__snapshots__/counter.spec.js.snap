// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe ser identico al Snapshot 1`] = `
<div>
  <h2>Counter</h2>
  <p>100 <sup>2</sup> 10000</p>
  <div class="buttons-container"><button>+1</button><button>-1</button></div>
</div>
`;
